src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -57,6 +80,8 @@ internal TestAssetInfo(DirectoryInfo root, string assetName)
                 _dataDirectory
             };
 
+            RemoveDataDirectoryIfAssetChanged();
```

Comment:
It's a good practice to avoid doing things that change the state of the world in a constructor. Can this call be deferred to a method invocation?